src/controller/CHIPDevice.h

```diff
@@ -80,11 +81,17 @@ struct ControllerDeviceInitParams
     Inet::InetLayer * inetLayer                         = nullptr;
     PersistentStorageDelegate * storageDelegate         = nullptr;
     Credentials::OperationalCredentialSet * credentials = nullptr;
+    SessionIDAllocator * idAllocator                    = nullptr;
 #if CONFIG_NETWORK_LAYER_BLE
     Ble::BleLayer * bleLayer = nullptr;
 #endif
 };
 
+class Device;
+
+typedef void (*OnDeviceConnected)(void * context, Device * device);
```

Comment:
nit: Prefer using to typedef.

`using OnDeviceConnected = void(void * context, Device * device);`